string_manipulation.rb

```diff
@@ -243,6 +256,9 @@ def palindrome_check(my_phrase)
   return true
 end
 
+# Time complexity is O(n), the loop will run n times depending on the size of my_phrase
+# Space complexity is O(1), no new space in memory is being allocated.
```

Comment:
Actually, you are allocating first and last. But they take up constant space regardless of the size of the input string passed and hence the space complexity is O(1)